waspc/cli/src/Wasp/Cli/Command/Build.hs

```diff
@@ -58,40 +73,65 @@ build = do
-    -- we're copying all files and folders necessary for the build into the .wasp/build directory.
-    -- We chose this approach for 0.12.0 (instead of building from the project root) because:
-    --   - The build context remains small (~1.5 MB vs ~900 MB).
-    --   - We don't risk copying possible secrets from the project root into the build context.
-    --   - The commands for building the project stay the same as before
-    --     0.12.0, which is good for both us (e.g., for fly deployment) and our
-    --     users  (no changes in CI/CD scripts).
-    -- For more details, read the issue linked above.
-    copyUserFilesNecessaryForBuild waspProjectDir buildDir = do
-      copyDirectory
-        (waspProjectDir </> srcDirInWaspProjectDir)
-        (buildDir </> castRel srcDirInWaspProjectDir)
+    prepareFilesNecessaryForDockerBuild waspProjectDir buildDir = runExceptT $ do
+      -- Until we implement the solution described in https://github.com/wasp-lang/wasp/issues/1769,
+      -- we're copying all files and folders necessary for the build into the .wasp/build directory.
+      -- We chose this approach for 0.12.0 (instead of building from the project root) because:
+      --   - The build context remains small (~1.5 MB vs ~900 MB).
+      --   - We don't risk copying possible secrets from the project root into the build context.
+      --   - The commands for building the project stay the same as before
+      --     0.12.0, which is good for both us (e.g., for fly deployment) and our
+      --     users  (no changes in CI/CD scripts).
+      -- For more details, read the issue linked above.
+      liftIO $
+        copyDirectory
+          (waspProjectDir </> srcDirInWaspProjectDir)
+          (buildDir </> castRel srcDirInWaspProjectDir)
+
+      liftIO $
+        copyDirectory
+          (waspProjectDir </> dotWaspDirInWaspProjectDir </> generatedCodeDirInDotWaspDir </> sdkRootDirInGeneratedCodeDir)
+          (buildDir </> sdkRootDirInGeneratedCodeDir)
+
+      let packageJsonInBuildDir = buildDir </> castRel packageJsonInWaspProjectDir
+      let packageLockJsonInBuildDir = buildDir </> castRel packageLockJsonInWaspProjectDir
+
+      liftIO $
+        copyFile
+          (waspProjectDir </> packageJsonInWaspProjectDir)
+          packageJsonInBuildDir
+
+      liftIO $
+        copyFile
+          (waspProjectDir </> packageLockJsonInWaspProjectDir)
+          packageLockJsonInBuildDir
 
-      copyDirectory
-        (waspProjectDir </> dotWaspDirInWaspProjectDir </> generatedCodeDirInDotWaspDir </> sdkRootDirInGeneratedCodeDir)
-        (buildDir </> sdkRootDirInGeneratedCodeDir)
+      -- A hacky quick fix for https://github.com/wasp-lang/wasp/issues/2368
+      -- We should remove this code once we implement a proper solution.
+      ExceptT $ updateJsonFile removeWaspConfigDevDependency packageJsonInBuildDir
+      ExceptT $ updateJsonFile removeAllMentionsOfWaspConfig packageLockJsonInBuildDir
```

Comment:
I originally had two functions which I called from the main body of the build function:
- `copyFilesNecessaryForDockerBuild` that copies everything.
- `editNpmFilesForDockerBuild` that edits package.json and package-lock.json.

This looked a little cleaner but forced me to define `packageJsonInBuildDir` and `packageLockJsonInBuildDir` twice, so I decided to stick with a single function (since the second one heavily relies on the first one being called beforehand, and they share some variables).

Reply:
Ok, yeah I guess it doesn't make a big difference. Although you could also have written helper functions that produce `packageJsonInBuildDir` and the other one for you when needed, to avoid duplication.

What I wonder more about is the name `prepareFilesNecesasryForDockerBuild`-> some of them are necessary for docker build, but that is not their only purpose, right? If we decided we are not producing a Dockerfile for users anymore, and they are supposed to write their own, could I just delete this whole function and would everything work fine, in the sense that all it is doing is only needed for Dockerfile? If so, then this name is fine. If not, I would ideally split logic into two pieces: first that copies over what needs to be copied in general, regardless if it will be used in Dockerfile or not, and then the other part is additional logic that is neccessary specifically in order to support Dockerfile.

Reply:
> Although you could also have written helper functions that produce packageJsonInBuildDir and the other one for you when needed, to avoid duplication.

I've considered this but decided it would fall into overengineering. Do you think it would be an improvement?

> If we decided we are not producing a Dockerfile for users anymore, and they are supposed to write their own, could I just delete this whole function and would everything work fine, in the sense that all it is doing is only needed for Dockerfile?

Yes, I believe that's the case. The only reason we're copying these files is because we need them in Docker's build context. **If we weren't using docker, we wouldn't need this function.** This is already documented at the [beginning of the function](https://github.com/wasp-lang/wasp/pull/2372/files#diff-ea4b4106c35cd1478bae7f04a9c2a368284228df2e37ae27aa4894b8c8f773c9R83), and we soon plan to remove it altogether.

Did I understand the question correctly?